java/vector/src/main/java/org/apache/arrow/vector/UInt2Vector.java

```diff
@@ -80,7 +80,7 @@ public MinorType getMinorType() {
    * @return value stored at the index.
    */
   public static char get(final ArrowBuf buffer, final int index) {
-    return buffer.getChar(index * TYPE_WIDTH);
+    return buffer.getChar((long) index * TYPE_WIDTH);
```

Comment:
would it be simpler to make TYPE_WIDTH long for here an other places?  Are there downsides to that?

Reply:
Making it a long is a reasonable choice. 
I am not sure if is an overkill as TYPE_WIDTH is often a small number, and often used in scenarios where a small number is expected.

Reply:
@emkornfield After more thoughts, I kind of prefer leaving the offset width int32. This makes sure we promote the related operations to int64 only when necessary. 

That being said, I will follow your advice and make the offset width int 64, if you have a reason, or feel strong about it.

Reply:
sounds ok to me.